test/pkg.jl

```diff
@@ -182,4 +183,37 @@ temp_pkg_dir() do
         finalize(repo)
         Pkg.update()
     end
+
+    #test PkgDev redirects
+    begin
+        try
+            Pkg.register("IDoNotExist")
```

Comment:
same with the 2 try-catches above that @jakebolewski added for `Pkg.test("IDoNotExist")` and `Pkg.test("IDoNotExist1", "IDoNotExist2")`.